packages/devtools_app/lib/src/profiler/cpu_profile_transformer.dart

```diff
@@ -50,6 +56,10 @@ class CpuProfileTransformer {
 
     // Use batch processing to maintain a responsive UI.
     while (_stackFramesProcessed < _stackFramesCount) {
+      if (processId != _activeProcessId) {
```

Comment:
nit: I think the best place for this throw is right after each `await` call as you will want to add more checks if you add more awaits.

Reply:
done